bundle-workflow/jenkins_workflow/integtest/Jenkinsfile

```diff
@@ -0,0 +1,69 @@
+pipeline {
+    agent none
+    environment {
+        BUNDLE_TEST_ID = "${BUILD_NUMBER}"
+    }
+    tools {
+        jdk "JDK14"
+        maven "maven-3.8.2"
+    }
+    stages {
+        stage('Parameters') {
+            steps {
+                script {
+                    properties([
+                            parameters([
+                                    string(
+                                            defaultValue: '',
+                                            name: 'bundle_manifest',
+                                            trim: true
+                                    ),
+                                    string(
+                                            defaultValue: '',
+                                            name: 'build_manifest',
+                                            trim: true
+                                    ),
+                                    string(
+                                            defaultValue: '',
+                                            name: 'architecture',
+                                            trim: true
+                                    ),
+                            ])
+                    ])
+                }
+            }
+        }
+        stage('Choose Agent Node') {
+            steps {
+                script {
+                    if ("${architecture}" == 'x64') {
+                        agentLabel = "Jenkins-Agent-al2-x64-m5xlarge"
+                    } else if ("${architecture}" == 'arm64') {
+                        agentLabel = "Jenkins-Agent-al2-arm64-m6gxlarge"
+                    } else {
+                        agentLabel = ""
+                    }
+                }
+            }
+        }
+        stage('Run Integration Tests') {
+            agent {
+                node {
+                    label "${agentLabel}"
+                }
+            }
+            steps {
+                withAWS(role: 'opensearch-test', roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
+                    s3Download(file: 'tests/bundle_manifest.yml', bucket: "${ARTIFACT_BUCKET_NAME}", path: "$bundle_manifest", force:true)
+                    s3Download(file: 'tests/build_manifest.yml', bucket: "${ARTIFACT_BUCKET_NAME}", path: "$build_manifest", force:true)
+                }
+                sh "./bundle-workflow/test_integration.sh --bundle-manifest tests/bundle_manifest.yml --build-manifest tests/build_manifest.yml --test-manifest bundle-workflow/src/test_workflow/config/test_manifest.yml"
```

Comment:
Could we reshape this workflow step to be more granular and avoid having to know about s3, here is what I was thinking it would looking like:
```
steps { 
   withAWS(role: "opensearch-integration-test") {
      s3Download(file: "tests/bundle_manifest.yml", bucket: "${ARTIFACT_BUCKET_NAME}", path: "$bundle_manifest", force: true)
      sh "./bundle-workflow/download_bundle --manifest $bundle_manifest --destination /bundle_under_test"
   }
   sh "./bundle-workflow/test_integration.sh --manifest $bundle_manifest --bundle_under_test /bundle_under_test" --test-artifacts /test_artifacts-$BUNDLE_TEST_ID
}
post() {
   always {
      withAWS(role: 'opensearch-integration-test', ...) {
         sh "./bundle-workflow/collect_additional_artifacts.sh --test-artifacts /test_artifacts-${BUNDLE_TEST_ID}"
         s3Upload(file: '/test_artifacts-${architecture}-${BUNDLE_TEST_ID}', bucket: '${test_results_bucket}')
      }
   }
}
```

Reply:
As we discussed, eventually all the tests will consume a place to store results and pull dependencies. 
Unfortunately we have to live with it that these test systems need to understand for now.

Eventually we should be able to publish test results from the jenkins stage avoiding the test suites to manage it for them  and be agnostic of where they are being stored.

Reply:
> Unfortunately we have to live with it that these test systems need to understand for now.

We can reduce a large amount of complexity in the tests if we do this.  If we are seeing trouble with the existing system please consider this an option to migrate to rather than additional investment

Reply:
@saratvemulapalli we don't need to download manifest files from s3 here. We are doing that as part of the individual test jobs. The jenkins job should supply - s3_bucket_name, opensearch_version, build_id, architecture and test_id to the underlying tool for the test run

Reply:
Absolutely, that was the discussion we had with @peternied. Agree that this should go to orchestrator and the jobs will just expect a path to these manifests.

Reply:
No, the orchestrator shouldn't be pulling these manifest files.

Since each of these jobs runs (can run) on a different host than orchestrator, the orchestrator (pipeline) shouldn't be pulling manifest files. Individual jobs should be pulling them when they run based in the input parameters - s3_bucket, opensearch_version, build_id, architecture. This logic is already added to Integ tests in #456

Reply:
Makes sense. So we are refactoring pulling the manifests into the Python scripts and keeping the jenkins job independent of it. 
 So I would have to wait until #456 is merged. 
 
 This would mean:
1. The Jenkins job will simply read through all the parameters which integtest needs to run and these parameters could be fed from test orchestrator vs :
    a. s3 bucket
    b. opensearch version
    c. build_id
    d. test_id
    e. architecture
2. Jenkins job will not pull the manifest files as the python test suites are pulling the manifests from S3. The consequence of this is that the Jenkins job will need to pull an architecture parameter which should be passed and will not read from the manifest.

Reply:
That's correct. Instead of orchestrator, we want individual test jobs to pull the manifest and other artifacts from S3 because the jobs can run on any host irrespective of where orchestrator logic runs. This keeps it decoupled.
 
For testing purpose, you could use feature branch of 456 to unblock this.